autogen.pl

```diff
@@ -1149,6 +1149,105 @@ sub in_tarball {
     return $tarball;
 }
 
+##############################################################################
+
+my %cached_config_timestamps;
+
+sub replace_config_sub_guess {
+    # This could be simpler if we could use some Perl modules for this
+    # functionality (e.g., DateTime).  But I don't want to introduce
+    # any CPAN dependencies here, so just do sometime simple, even if
+    # it's a bit laborious. Use a few private helper functions for
+    # this kind of functionality.
+
+    sub _get_timestamp {
+        my $filename = shift;
+
+        my $ret;
+        if (-x $filename) {
+            my $out = `$filename --version`;
+            $out =~ m/GNU config\.[a-z]+ \((.+)\)/;
+            $ret = $1;
+        }
+
+        return $ret;
+    }
+
+    sub _split_timestamp {
+        my $ts = shift;
+
+        $ts =~ m/(\d\d\d\d)-(\d\d)-(\d\d)/;
+        return $1, $2, $3;
+    }
+
+    # Returns true if timestamp $a > timestamp $b.
+    sub _timestamp_gt {
+        my ($a, $b) = @_;
+
+        my ($year_a, $month_a, $day_a) = _split_timestamp($a);
+        my ($year_b, $month_b, $day_b) = _split_timestamp($b);
+
+        # Don't try to be clever -- just do a simple set of explicit
+        # comparisons.
+        if ($year_a > $year_b) {
+            return 1;
+        } elsif ($year_a < $year_b) {
+            return 0;
+        } else {
+            if ($month_a > $month_b) {
+                return 1;
+            } elsif ($month_a < $month_b) {
+                return 0;
+            } else {
+                if ($day_a > $day_b) {
+                    return 1;
+                } else {
+                    return 0;
+                }
+            }
+        }
+    }
+
+    my ($topdir) = @_;
+
+    # Find the stashed known-good files, and get their version
+    # timestamps.
+    my $cached_dir = "$topdir/config/from-savannah";
+    my @files = qw/config.guess config.sub/;
+    foreach my $file (@files) {
+        if (!exists($cached_config_timestamps{$file})) {
+            my $filename = "$cached_dir/upstream-$file";
+            my $ts = _get_timestamp($filename);
+            $cached_config_timestamps{$file} = $ts
+                if ($ts);
+        }
+    }
+
+    # If we get here and there's not 2 stashed timestamps, that means
+    # we don't have the two stashed files.  So just use whatever
+    # Autoconf installed.
+    return
+        if (2 != keys(%cached_config_timestamps));
+
+    # Find all config.guess/config.sub files in the tree.  If their
+    # versions are older than the stashed known-good files, update
+    # them from the stash.
+    my @files;
+    File::Find::find(sub {
```

Comment:
oh.  hum.  I'm not sure this is a good idea.  This will update files in prrte and pmix, for example.  Worse, it will *sometimes* update files for libevent.  On a fresh tarball, libevent will not have been expanded by this point, but if the tarball has already been built once, the libevent tarball will already be expaneded and the files will be replaced.

Reply:
That was actually my intention: to replace `config.*` throughout the entire tree (e.g., it picks up the `config.*` files in ROMIO this way).

I don't think that I knew (remembered?) that we're shipping tarballs in our tarball.  Hrm.  Should I just have it skip the 3rd party directory?

Reply:
i don't have a good rubric here.  You're right that we should update ROMIO.  Assuming the PMIX org pulls in the change for pmix and prrte, we could have competing updates.  So we should talk to Ralph about integration plans (particularly for 4.0 and 4.1 branches.  But regardless, we should skip `3rd-party/libevent-*`.